flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/DescribeModelOperation.java

```diff
@@ -0,0 +1,156 @@
+ * statement.
+ */
+@Internal
+public class DescribeModelOperation implements Operation, ExecutableOperation {
+
+    private final ObjectIdentifier sqlIdentifier;
+    private final boolean isExtended;
+
+    public DescribeModelOperation(ObjectIdentifier sqlIdentifier, boolean isExtended) {
+        this.sqlIdentifier = sqlIdentifier;
+        this.isExtended = isExtended;
+    }
+
+    public ObjectIdentifier getSqlIdentifier() {
+        return sqlIdentifier;
+    }
+
+    public boolean isExtended() {
+        return isExtended;
+    }
+
+    @Override
+    public String asSummaryString() {
+        Map<String, Object> params = new LinkedHashMap<>();
+        params.put("identifier", sqlIdentifier);
+        params.put("isExtended", isExtended);
+        return OperationUtils.formatWithChildren(
+                "DESCRIBE MODEL", params, Collections.emptyList(), Operation::asSummaryString);
+    }
+
+    @Override
+    public TableResultInternal execute(Context ctx) {
+        // DESCRIBE MODEL <model> shows input/output schema if any.
+        Optional<ContextResolvedModel> result = ctx.getCatalogManager().getModel(sqlIdentifier);
+        if (result.isEmpty()) {
+            throw new ValidationException(
+                    String.format(
+                            "Model with the identifier '%s' doesn't exist.",
+                            sqlIdentifier.asSummaryString()));
+        }
+
+        ResolvedSchema inputSchema = result.get().getResolvedModel().getResolvedInputSchema();
+        ResolvedSchema outputSchema = result.get().getResolvedModel().getResolvedOutputSchema();
+        Object[][] rows = generateModelColumnsRows(inputSchema, outputSchema);
+
+        boolean nonComments = isSchemaNonColumnComments(inputSchema, outputSchema);
+        return buildTableResult(
+                generateTableColumnsNames(nonComments),
+                generateTableColumnsDataTypes(nonComments),
+                rows);
+    }
+
+    private Object[][] generateModelColumnsRows(
+            ResolvedSchema inputSchema, ResolvedSchema outputSchema) {
+        boolean nonComments = isSchemaNonColumnComments(inputSchema, outputSchema);
+
+        return Streams.concat(
+                        inputSchema.getColumns().stream()
+                                .map((c) -> buildSingleRow(c, nonComments, true)),
+                        outputSchema.getColumns().stream()
+                                .map((c) -> buildSingleRow(c, nonComments, false)))
+                .toArray(Object[][]::new);
+    }
+
+    private Object[] buildSingleRow(Column c, boolean nonComments, boolean isInput) {
+        final LogicalType logicalType = c.getDataType().getLogicalType();
+        final ArrayList<Object> result =
+                new ArrayList<>(
+                        Arrays.asList(
+                                c.getName(),
+                                logicalType.copy(true).asSummaryString(),
```

Comment:
do we always want to have forced nullability here?

Reply:
Next column is specifically to indicate whether it can be nullable. So we don't display `not null` here